### ISSUE TITLE
pyramid_mailer.testing doesn't add mailer to the request object

### ISSUE THREAD

**Issue by dfee** [Pylons/pyramid_mailer#79]
As stated in the title, using `pyramid.includes = pyramid_mailer.testing` won't add the `mailer` attribute to a request object, while `pyramid.includes = pyramid_mailer` does add the attribute.

This almost seems like an oversight, except that that [testing.py](https://github.com/Pylons/pyramid_mailer/blob/master/pyramid_mailer/testing.py) was last touched almost six years ago (and I can't find a history of anyone complaining about this).

What was the reasoning behind this?

**Comment by dfee** [comment]
For anyone who comes this way, what I did was remove `pyramid.includes = pyramid_mailer` from my `*.ini` files and re-implemented the `includeme` to get the benefit of having `request.mailer` regardless of whether we're testing or not.

To discriminate a testing environment, set `mail.testing = true` in your `test.ini` file. If `mail.testing` is not set in your `*.ini` file, a `pyramid_mailer.mailer.Mailer` instance will be attached to `request.mailer` instead of a `pyramid_mailer.mailer.DummyMailer` instance.

As with the usual `pyramid.includes = pyramid_mailer`, you will be able to take advantage of the `get_mailer` function.

```
from pyramid.settings import asbool
from pyramid_mailer import (
    get_mailer,
    mailer_factory_from_settings,
)
from pyramid_mailer.interfaces import IMailer
from pyramid_mailer.mailer import DummyMailer


def includeme(config):
    """
    For some reason, pyramid_mailer.testing doesn't attach the `mailer` to the
    request. This solves that problem.
    """
    settings = config.registry.settings
    testing = asbool(settings.get('mail.testing', False))

    if testing:
        mailer = DummyMailer()
    else:
        mailer = mailer_factory_from_settings(settings)

    config.registry.registerUtility(mailer, IMailer)

    config.add_request_method(
        lambda r: mailer if testing else get_mailer(r),
        'mailer',
        reify=True
    )
```

**Comment by mmerickel** [comment]
Would you be interested in submitting a PR that adds the request method to the testing includeme and an associated test?

**Comment by mmerickel** [comment]
Also, I'd be happy to accept a PR for this `mail.testing` attribute, which I think is better than `config.include('pyramid_mailer.testing')`.

And finally, adding the `bind` method to the `DummyMailer` would also be a good idea.

**Comment by mmerickel** [comment]
Closed via #83.
